src/poetry/installation/installer.py

```diff
@@ -204,14 +203,14 @@ def _do_refresh(self) -> int:
         ):
             ops = solver.solve(use_latest=[]).calculate_operations()
 
-        local_repo = Repository()
+        local_repo = LockfileRepository()
```

Comment:
I think the introduction of a new type here makes the code/intention a fair bit more obvious as an ancillary benefit. I think we should probably go ahead and change the use of `local_repo` to `lockfile_repo` in this file to capitalize on that further.